batch/batch/front_end/front_end.py

```diff
@@ -1489,6 +1499,89 @@ async def insert(tx):
     set_message(session, f'Added billing project {billing_project}.', 'info')
     return web.HTTPFound(deploy_config.external_url('batch', '/billing_projects'))
 
+async def _close_billing_project(db, billing_project):
+    @transaction(db)
+    async def close_project(tx):
+        row = await tx.execute_and_fetchone(
+            '''
+SELECT name, closed,
+  (SELECT id FROM batches WHERE 
+    billing_project = billing_projects.name AND 
```

Comment:
I think we need a new data structure for this that tracks the number of open batches, running batches, etc. per billing project. This will get updated with a trigger. Otherwise, maybe an index on the batches table with billing_project and state might do it, but I'm worried that will be slow. Not sure.

Reply:
Hmm, good call. Do we need to do something similar for generating billing reports? Can I use what we're doing there?

I don't think we need this query to be super fast, since it should be rarely called and only on billing projects without any new activity, but definitely querying the entire `batches` table seems like a bad plan as it grows.

Reply:
I could also rewrite this as `SELECT id FROM batches WHERE time_completed IS NULL and billing_project = billing_projects.name`, right? That should take advantage of the index on `time_completed` while giving the same result?